optuna/samplers/_tpe/probability_distributions.py

```diff
@@ -0,0 +1,136 @@
+from typing import List
+from typing import NamedTuple
+from typing import TYPE_CHECKING
+from typing import Union
+
+import numpy as np
+
+from optuna._imports import _LazyImport
+
+
+if TYPE_CHECKING:
+    import scipy.special as special
+    import scipy.stats as stats
+else:
+    special = _LazyImport("scipy.special")
+    stats = _LazyImport("scipy.stats")
+
+
+class _BatchedCategoricalDistributions(NamedTuple):
+    weights: np.ndarray
+
+
+class _BatchedTruncNormDistributions(NamedTuple):
+    mu: np.ndarray
+    sigma: np.ndarray
+    low: float  # Currently, low and high do not change per trial.
+    high: float
+
+
+class _BatchedDiscreteTruncNormDistributions(NamedTuple):
+    mu: np.ndarray
+    sigma: np.ndarray
+    low: float  # Currently, low, high and step do not change per trial.
+    high: float
+    step: float
+
+
+_BatchedDistributionUnion = Union[
```

Comment:
This would be simple name.
```suggestion
_BatchedDistributions = Union[
```